### PR TITLE
Recreate security cert during vserver create.

### PR DESCRIPTION
**Pull request by kpawar-sap**
The certificate is automatically created on NetApp with 1 year i.e. 365 days of expiration time, and admin needs to manually extend it. It would be nice Manila can take care to create certs with admin configuable expiration time. Manila can first delete the cert, and then create a new cert with given expiration time while creating share server.

### CODE REVIEW

File: manila/share/drivers/netapp/options.py [hunk 1]
```diff
@@ -193,7 +193,15 @@
                 help='This option enables the logical space reporting on a '
                      'newly created vserver and locical space accounting '
                      'on newly created volumes on this vserver. ',
-                default=False), ]
+                default=False),
+    cfg.IntOpt('netapp_security_cert_expire_days',
+               min=1,
+               max=3652,
+               default=2190,
```
Comment (Carthaca) [inline manila/share/drivers/netapp/options.py:200]: Internally this is okay - but I think upstream it should match the NetApp default of 365 days here

File: manila/share/drivers/netapp/dataontap/client/client_cmode.py [hunk 2]
```diff
@@ -276,6 +280,55 @@ def modify_vserver(self, vserver_name, aggregate_names, retention_hours):
         }
         self.send_request('vserver-modify', modify_args)
 
+    @na_utils.trace
+    def _modify_security_cert_expire_days(self, vserver_name,
```
Comment (Carthaca) [inline manila/share/drivers/netapp/dataontap/client/client_cmode.py:284]: I suggest to go with a more general name `_modify_security_cert()` , expire-days is one of potentially many options this function may have in the future

**Review by Carthaca — CHANGES_REQUESTED** [review]
Please use the safer approach: only delete if new cert had been assigned (like outlined in the attachment of the jira ticket)

I.e.

1. create new cert
2. modify ssl of vserver to use new cert
3. delete old cert

File: manila/share/drivers/netapp/dataontap/client/client_cmode.py [hunk 3]
```diff
@@ -276,6 +280,55 @@ def modify_vserver(self, vserver_name, aggregate_names, retention_hours):
...
+            'type': 'server',
+            'vserver': vserver_name,
+        }
+        self.send_request('security-certificate-delete', delete_args)
+
+        create_args = {
+            'vserver': vserver_name,
+            'common-name': vserver_name,
+            'type': 'server',
+            'size': '2048',
```
[the remark this answers is not in the document]
Comment (kpawar-sap) [reply manila/share/drivers/netapp/dataontap/client/client_cmode.py:326]: ok

File: manila/share/drivers/netapp/dataontap/client/client_cmode.py (hunk 2 above)
Comment (kpawar-sap) [reply manila/share/drivers/netapp/dataontap/client/client_cmode.py]: fixed

**Comment by kpawar-sap** [comment]
SSL show gives expected output after recent pushed changes.

**Pull request merged**